R/git.R

```diff
@@ -102,14 +56,21 @@ git_user <- function(path) {
 #' @return The git user.email.
 #'
 #' @keywords internal
-git_email <- function(path) {
+git_email <- function(path = ".") {
   git_config(path, "user.email")
 }
 
 git_config <- function(path, field) {
-  args <- c("config", field)
-  res <- git_run(args, root = path, check = TRUE)
-  res$output
+  gitres <- gert::git_config(repo = git_root_directory(path))
+  gitres <- rbind(
+    gitres[gitres$level == 'local', ],
+    gitres[gitres$level != 'local', ])
+
+  if (!any(gitres$name == field)) {
+    stop("git config field not found")
```

Comment:
Could we return the `field` and `path` they asked for here? Thinking about if this is thrown when we're several layers deep in the call stack - more context is always useful. Or if we go and call this in future from somewhere where the calling method has a less clear name.

Reply:
Done